src/core/include/metaphysicl/ct_types.h

```diff
@@ -29,6 +29,9 @@
 #ifndef METAPHYSICL_CT_TYPES_H
 #define METAPHYSICL_CT_TYPES_H
 
+#include <array>
+#include <vector>
```

Comment:
There's no way to avoid this, but I hate this.

C++ really needs some facility for letting Foo.h have an optional "specializations related to Bar" section that only activates if Bar.h is also included elsewhere.  The current alternatives seem to be "you create FooBar.h, and if you add Baz specializations you create BarBaz.h and FooBaz.h too, and if you're lucky you don't need FooBarBaz.h and your complexity will only be quadratic not exponential", or "whatever, just include Baz.h whether users need it or not, everything includes everything else, you might as well not even bother with non-Unity builds or modularity".